File: src/model/usermodel.ts
import { User } from "@prisma/client"

export interface RegisterUserRequest {
    username: string
    email: string
    password: string
}

export interface UserResponse {
    token?: string
    username: string
}

export interface LoginUserRequest {
    email: string
    password: string
}

export function toUserResponse(prismaUser: User): UserResponse {
    return {
        token: prismaUser.token ?? "",
        username: prismaUser.username,
    }
}

File: models/UserModel.kt
package com.example.alp_vp.models

data class RegisterUserRequest(
    val username: String,
    val email: String,
    val password: String
)

data class UserResponse(
    val token: String?,
    val username: String
)

data class LoginUserRequest(
    val email: String,
    val password: String
)

fun toUserResponse(user: User): UserResponse {
    return UserResponse(
        token = user.token ?: "",
        username = user.username
    )
}

data class User(
    val token: String?,
    val username: String
)
